ui/store/actions.js

```diff
@@ -2712,10 +2727,21 @@ export async function updateTokenType(tokenAddress) {
   return token;
 }
 
+/**
+ * initiates polling for gas fee estimates.
+ *
+ * @returns {string} a unique identify of the polling request that can be used
+ *  to remove that request from consideration of whether polling needs to
+ *  continue.
+ */
 export function getGasFeeEstimatesAndStartPolling() {
   return promisifiedBackground.getGasFeeEstimatesAndStartPolling();
 }
 
+export function disconnectGasFeeEstimatePoller(pollToken) {
```

Comment:
@darkwing this will be helpful for you, this is what should be called in the return method of that useEffect hook so that it stops polling.